libwallet/src/internal/selection.rs

```diff
@@ -65,20 +69,25 @@ where
 		change_outputs,
 		selection_strategy_is_use_all,
 		&parent_key_id,
+		include_inputs_in_sum,
 	)?;
 
 	// Update the fee on the slate so we account for this when building the tx.
 	slate.fee = fee;
 
-	let blinding = slate.add_transaction_elements(keychain, &ProofBuilder::new(keychain), elems)?;
+	let blinding = match elems.len() {
+		// in the event of no change, keep our blinding factor at zero.
+		// it will be adjusted when we add inputs
+		0 => BlindingFactor::zero(),
+		_ => slate.add_transaction_elements(keychain, &ProofBuilder::new(keychain), elems)?,
+	};
```

Comment:
Why can't we pass empty elems to add_transaction_elements() ? (which I would also expect to yield BlindingFactor::zero())

Reply:
Yes, I can move the check into `add_transaction_elements`, but the fundamental issue is the `build` module (contained in grin core) throws the error if elems is empty. The `build` library is some of the the oldest untouched part of the code and could perhaps do with another revision at some stage.